underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystemFactory.java

```diff
@@ -79,11 +82,9 @@ public boolean supportsPath(String path, UnderFileSystemConfiguration conf) {
       // are supported this is not an option and we have to continue to use this method.
       for (final String prefix : alluxioConf.getList(PropertyKey.UNDERFS_HDFS_PREFIXES, ",")) {
         if (path.startsWith(prefix)) {
-          if (conf != null && HdfsVersion
-              .find(conf.get(PropertyKey.UNDERFS_VERSION)) != HdfsVersion.find(getVersion())) {
-            continue;
+          if (conf == null || HdfsVersion.matches(conf.get(PropertyKey.UNDERFS_VERSION), getVersion())) {
+            return true;
```

Comment:
I don't think this logic is correct? We don't want to return true if the configuration is null.

Reply:
I spoke to @madanadit and he mentioned that the logic was if the config was null then all jars should support the path. It does change the logic from before and I am not sure which one makes more sense.

Reply:
The logic is identical to before, we just flipped the condition and switched the looping logic